sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeFileSystemClient.cs

```diff
@@ -2590,7 +2590,12 @@ private void SetNameFieldsIfNull()
         /// A <see cref="Exception"/> will be thrown if a failure occurs.
         /// </remarks>
         public virtual Uri GenerateSasUri(DataLakeFileSystemSasPermissions permissions, DateTimeOffset expiresOn) =>
-            GenerateSasUri(new DataLakeSasBuilder(permissions, expiresOn) { FileSystemName = Name });
+            GenerateSasUri(
+                new DataLakeSasBuilder(permissions, expiresOn)
+                {
+                    FileSystemName = Name,
+                    Version = StorageVersionExtensions.ToVersionString(ClientConfiguration.Version)
```

Comment:
This is the other customer-facing change.  Now, the SAS version will match the x-ms-version of the client that generated it.  This is necessary to pass the correct sas version into our tests.

Reply:
I also made this change in other client's `GenerateSasUri()` methods.